server/protocols/webadmin/webadmin-cassandra/src/main/java/org/apache/james/webadmin/routes/RunningOptionsParser.java

```diff
@@ -0,0 +1,44 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.routes;
+
+import java.util.Optional;
+
+import org.apache.james.mailbox.cassandra.mail.task.SolveMessageInconsistenciesService.RunningOptions;
+
+import spark.Request;
+
+public class RunningOptionsParser {
+    public static RunningOptions parse(Request request) {
+        return intQueryParameter(request, "messagesPerSecond")
+            .map(RunningOptions::new)
+            .orElse(RunningOptions.DEFAULT);
+    }
+
+    public static Optional<Integer> intQueryParameter(Request request, String queryParameter) {
+        try {
+            return Optional.ofNullable(request.queryParams(queryParameter))
+                .map(Integer::parseInt);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(String.format("Illegal value supplied for query parameter '%s', expecting a " +
+                "strictly positive optional integer", queryParameter), e);
```

Comment:
s/a strictly positive optional integer/an optional strictly positivel integer"

i don't see the strictly positive validation here.

Reply:
That validation is done at the tasklayer.

The webadmin layer tests this case, I don't see why we would need to duplicate a checkalready done further in the task layer. Could that be considered coupling?

Reply:
Ok, it's just that by looking only at this piece of code the error message seemed to not be in accordance with the code.
In scala i would told you to do the validation here and return a refined type. Then use this type in parameter of the task.

For me the running option parser should accept only valid value. Which is not the case here.
I think here strong typing the concurrency could be a nice thing

Reply:
It end up generating only valid value, just `negative/zero` rejection is performed by the RunningOption class checks.

I don't see added value in duplicating the check.